kokain-ksp/src/main/java/com/schwarz/kokain/ksp/validation/PreValidator.kt

```diff
@@ -0,0 +1,80 @@
+package com.schwarz.kokain.ksp.validation
+
+import com.google.devtools.ksp.KspExperimental
+import com.google.devtools.ksp.getJavaClassByName
+import com.google.devtools.ksp.getKotlinClassByName
+import com.google.devtools.ksp.processing.KSPLogger
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.google.devtools.ksp.symbol.KSType
+import com.google.devtools.ksp.symbol.Modifier
+import com.schwarz.kokain.api.EBean
+import com.schwarz.kokain.api.EFactory
+import com.schwarz.kokain.api.KDiFactory
+import com.schwarz.kokain.ksp.util.getKSAnnotationsByType
+import com.schwarz.kokain.ksp.util.getKSValueArgumentByName
+import com.squareup.kotlinpoet.ksp.toTypeName
+
+class PreValidator(logger: KSPLogger, val resolver: Resolver) {
+
+    private val logger = logger
+
+    @Throws(ClassNotFoundException::class)
+    fun validateEbean(model: KSClassDeclaration): Boolean {
+
+        val entityElement = model
+        var result = true
+
+        if (entityElement.modifiers.contains(Modifier.PRIVATE)) {
+            logger.error(EBean::class.java.simpleName + " can not be private", entityElement)
+            result = false
+        }
+        if (entityElement.modifiers.contains(Modifier.PROTECTED)) {
```

Comment:
is this the entire class? or the individual properties?
a protected property should be fine, shouldn't it?

Reply:
This checks for the class visibility and basically the generated factory which are generated will not be able to see this protected class.